sw/STM32CubeIDE_Project/Core/Src/dcf77Decoding.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * dcf77Decoding.c
+ *
+ *  Created on: Sep 9, 2024
+ *      Author: Dzale
+ */
+
+#include "dcf77Decoding.h"
+
+/* Constants related to DCF77 protocol */
+/* Min and max values from da6180B.pdf */
+const uint16_t ZERO_PULS_DURATION_MAX = 130;
+const uint16_t ZERO_PULS_DURATION_MIN = 40;
+const uint16_t ONE_PULS_DURATION_MAX = 250;
+const uint16_t ONE_PULS_DURATION_MIN = 140;
+
+const uint16_t MINUTE_MARK_PULS_DURATION_MAX = 2100;
+const uint16_t MINUTE_MARK_PULS_DURATION_MIN = 1600;
+
+enum PulseType DCF77_CheckPulseType(DCF77_TimeSample_t* sampleToCheck)
+{
+	if(		sampleToCheck->pulseLength > ZERO_PULS_DURATION_MIN && \
+			sampleToCheck->pulseLength < ZERO_PULS_DURATION_MAX)
+	{
+		// Special case on 58th pulse is that is always 0 and that minute mark is determined by signal Length
```

Comment:
Since you consider a signal length from falling edge to falling edge this won't be 58th pulse, but first one